xalia/Win32/HwndListViewProvider.cs

```diff
@@ -252,6 +261,82 @@ protected override async Task<int> FetchItemCount()
             }
             return Utils.TruncatePtr(result);
         }
+
+        public override string[] GetTrackedProperties()
+        {
+            return tracked_properties;
+        }
+
+        public override void TrackedPropertyChanged(UiDomElement element, string name, UiDomValue new_value)
+        {
+            if (name == "recurse_method")
+            {
+                string string_value = new_value is UiDomString id ? id.Value : string.Empty;
+                switch (string_value)
+                {
+                    case "win32_list":
+                        if (watching_children && !watching_children_visible)
+                            break;
+                        watching_children = true;
+                        watching_children_visible = false;
+                        WatchChildren();
+                        break;
+                    case "win32_list_visible":
+                        if (watching_children && watching_children_visible)
+                            break;
+                        watching_children = true;
+                        watching_children_visible = true;
+                        WatchChildren();
+                        break;
+                    default:
+                        watching_children = false;
+                        UnwatchChildren();
+                        break;

```

Comment:
Is there some value we could set `watching_children_visible` to here as a way to signify it has been cleared? E.g, if we hit this default case with `watching_children_visible` set to true, and then hit `case "win32_list_visible":` afterwards with `watching_children_visible` set to true again, it will early exit without calling `WatchChildren()`.

Reply:
I don't think it will. If it hits the `"win32_list_visible"` case while `watching_children` is false, then the if check will be false and it will continue to the rest of the case, regardless of the previous value of `watching_children_visible`.

Reply:
Oh right, doh. Disregard this. :)